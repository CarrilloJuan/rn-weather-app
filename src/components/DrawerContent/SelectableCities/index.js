import React from 'react';
import {useDispatch} from 'react-redux';
import {DrawerItem} from '@react-navigation/drawer';
import {FlatList, View, StyleSheet} from 'react-native';

import {fetchWeatherByCity} from '../../../store/weatherSlice';

const styles = StyleSheet.create({
  container: {
    paddingLeft: 24,
  },
});

const renderItem = (item, dispatch) => (
  <DrawerItem
    testID="city-item"
    label={item.cityName}
    onPress={() => dispatch(fetchWeatherByCity(item))}
  />
);

export default function SelectableCities({cities}) {
  const dispatch = useDispatch();

  return (
    <View style={styles.container}>
      <FlatList
        data={cities}
        renderItem={({item}) => renderItem(item, dispatch)}
        keyExtractor={(item) => `${item.id}`}
      />
    </View>
  );
}
