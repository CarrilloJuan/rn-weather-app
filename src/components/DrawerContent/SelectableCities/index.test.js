/**
 * @jest-environment jsdom
 */

import React from 'react';
import {shallow, mount} from 'enzyme';
import {useDispatch} from 'react-redux';
import SelectableCities from '.';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

describe('Drawer Component', () => {
  // Remove warnings when rendering react-native components
  const origConsole = console.error;
  beforeEach(() => {
    console.error = () => {};
  });

  afterEach(() => {
    console.error = origConsole;
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('renders as expected', () => {
    const wrapper = shallow(<SelectableCities />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call the dispatch when onPress', () => {
    const citiesMock = [{cityName: 'City', id: '3646738'}];
    const wrapper = mount(<SelectableCities cities={citiesMock} />);
    const mockedDispatch = jest.fn();
    useDispatch().mockReturnValue(mockedDispatch);
    wrapper.find({testID: 'city-item'}).at(0).props().onPress();
    expect(mockDispatch.mock.calls.length).toBe(1);
  });
});
