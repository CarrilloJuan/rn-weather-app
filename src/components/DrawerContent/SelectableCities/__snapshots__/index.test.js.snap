// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Component renders as expected 1`] = `
<View
  style={
    Object {
      "paddingLeft": 24,
    }
  }
>
  <FlatList
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
